ern-cauldron-api/src/CauldronApi.ts

```diff
@@ -803,6 +803,70 @@ export default class CauldronApi {
   // FILE OPERATIONS
   // =====================================================================================
 
+  // -------------------------------------------------------------------------------------
+  // ARBITRARY FILE ACCESS
+  // -------------------------------------------------------------------------------------
+
+  public async addFile({
+    cauldronFilePath,
+    fileContent,
+    fileMode,
+  }: {
+    cauldronFilePath: string
+    fileContent: string | Buffer
+    fileMode?: string
+  }) {
+    if (!cauldronFilePath) {
+      throw new Error('[addFile] cauldronFilePath is required')
+    }
+    if (!fileContent) {
+      throw new Error('[addFile] fileContent is required')
+    }
+    if (await this.hasFile({ cauldronFilePath })) {
+      throw new Error(
+        `[addFile] ${cauldronFilePath} already exists. Use updateFile instead.`
+      )
+    }
+    return this.fileStore.storeFile(cauldronFilePath, fileContent, fileMode)
+  }
+
+  public async updateFile({
+    cauldronFilePath,
+    fileContent,
+    fileMode,
+  }: {
+    cauldronFilePath: string
+    fileContent: string | Buffer
+    fileMode?: string
+  }) {
+    if (!cauldronFilePath) {
+      throw new Error('[updateFile] cauldronFilePath is required')
+    }
+    if (!fileContent) {
+      throw new Error('[updateFile] fileContent is required')
+    }
+    if (!(await this.hasFile({ cauldronFilePath }))) {
```

Comment:
:minor: extra ()

Reply:
That's the way it is across the code base. Not intended but no real choice, this is due to `prettier` auto formatting of the source code. Someone reported this in https://github.com/prettier/prettier/issues/746 but in the end I think decision was to keep them.